cdap-ui/app/features/tracker/services/my-lineage-service.js

```diff
@@ -0,0 +1,214 @@
+        label: data.name,
+        id: key,
+        nodeType: 'data',
+        entityId: data.name,
+        entityType: data.type,
+        displayType: data.displayType,
+        icon: data.icon
+      };
+
+      uniqueNodes[key] = nodeObj;
+    });
+
+
+    /* SETTING CONNECTIONS */
+    angular.forEach(response.relations, (rel) => {
+      let isUnknownOrBoth = rel.access === 'both' || rel.access === 'unknown';
+
+      if (rel.access === 'read' || isUnknownOrBoth) {
+        let dataId = rel.data === currentActiveNode ? rel.data : rel.data + '-read';
+        let programId = rel.data === currentActiveNode ? rel.program + '-read' : rel.program + '-write';
+        connections.push({
+          source: dataId,
+          target: programId,
+          type: 'read'
+        });
+
+        nodes.push({
+          dataId: dataId,
+          uniqueNodeId: rel.data,
+          isLeftEdge: rel.data !== currentActiveNode
+        });
+        nodes.push({
+          dataId: programId,
+          uniqueNodeId: rel.program
+        });
+      }
+
+      if (rel.access === 'write' || isUnknownOrBoth) {
+        let dataId = rel.data === currentActiveNode ? rel.data : rel.data + '-write';
+        let programId = rel.data === currentActiveNode ? rel.program + '-write' : rel.program + '-read';
+        connections.push({
+          source: programId,
+          target: dataId,
+          type: 'write'
+        });
+
+        nodes.push({
+          dataId: dataId,
+          uniqueNodeId: rel.data,
+          isRightEdge: rel.data !== currentActiveNode
+        });
+        nodes.push({
+          dataId: programId,
+          uniqueNodeId: rel.program
+        });
+      }
+
+      uniqueNodes[rel.program].runs = uniqueNodes[rel.program].runs.concat(rel.runs);
+      uniqueNodes[rel.program].runs = _.uniq(uniqueNodes[rel.program].runs);
+    });
+
+    nodes = _.uniq(nodes, (n) => { return n.dataId; });
+    let graph = this.getGraphLayout(nodes, connections);
+    this.mapNodesLocation(nodes, graph);
+
+    return {
+      connections: connections,
+      nodes: nodes,
+      uniqueNodes: uniqueNodes,
+      graph: graph
+    };
+  }
+
+  parseProgramType(programType) {
+    switch (programType) {
+      case 'Flow':
+        return 'flows';
+      case 'Mapreduce':
+        return 'mapreduce';
+      case 'Spark':
+        return 'spark';
+      case 'Worker':
+        return'workers';
+      case 'Workflow':
+        return 'workflows';
+      case 'Service':
+        return 'services';
+    }
+  }
+
+  parseDataInfo(data) {
+    if (data.entityId.type === 'datasetinstance') {
+      return {
+        name: data.entityId.id.instanceId,
+        type: 'datasets',
+        icon: 'icon-datasets',
+        displayType: 'Dataset'
+      };
+    } else {
+      return {
+        name: data.entityId.id.streamName,
+        type: 'streams',
+        icon: 'icon-streams',
+        displayType: 'Stream'
+      };
+    }
+  }
+
+  getProgramIcon(programType) {
```

Comment:
We can, we just need to inject that service to wherever we want to use that function